flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adapter/DefaultResultPartition.java

```diff
@@ -0,0 +1,111 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.scheduler.adapter;
+
+import org.apache.flink.runtime.io.network.partition.ResultPartitionType;
+import org.apache.flink.runtime.jobgraph.IntermediateDataSetID;
+import org.apache.flink.runtime.jobgraph.IntermediateResultPartitionID;
+import org.apache.flink.runtime.scheduler.strategy.SchedulingExecutionVertex;
+import org.apache.flink.runtime.scheduler.strategy.SchedulingResultPartition;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import static org.apache.flink.runtime.scheduler.strategy.SchedulingResultPartition.ResultPartitionState.DONE;
+import static org.apache.flink.runtime.scheduler.strategy.SchedulingResultPartition.ResultPartitionState.EMPTY;
+import static org.apache.flink.runtime.scheduler.strategy.SchedulingResultPartition.ResultPartitionState.PRODUCING;
+import static org.apache.flink.runtime.scheduler.strategy.SchedulingResultPartition.ResultPartitionState.RELEASED;
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+/**
+ * Default implementation of {@link SchedulingResultPartition}.
+ */
+public class DefaultResultPartition implements SchedulingResultPartition {
+
+	private final IntermediateResultPartitionID resultPartitionId;
+
+	private final IntermediateDataSetID intermediateDataSetId;
+
+	private final ResultPartitionType partitionType;
+
+	private SchedulingExecutionVertex producer;
+
+	private final List<SchedulingExecutionVertex> consumers;
+
+	DefaultResultPartition(
+			IntermediateResultPartitionID partitionId,
+			IntermediateDataSetID intermediateDataSetId,
+			ResultPartitionType partitionType) {
+		this.resultPartitionId = checkNotNull(partitionId);
+		this.intermediateDataSetId = checkNotNull(intermediateDataSetId);
+		this.partitionType = checkNotNull(partitionType);
+		this.consumers = new ArrayList<>();
+	}
+
+	@Override
+	public IntermediateResultPartitionID getId() {
+		return resultPartitionId;
+	}
+
+	@Override
+	public IntermediateDataSetID getResultId() {
+		return intermediateDataSetId;
+	}
+
+	@Override
+	public ResultPartitionType getPartitionType() {
+		return partitionType;
+	}
+
+	@Override
+	public ResultPartitionState getState() {
+		switch (producer.getState()) {
+			case CREATED:
+			case SCHEDULED:
+			case DEPLOYING:
+				return EMPTY;
+			case RUNNING:
+				return PRODUCING;
+			case FINISHED:
+				return DONE;
+			default:
+				return RELEASED;
```

Comment:
Was it a conscious decision to return `RELEASED` as the default? Since `RELEASED` is currently not implemented, I'd rather not see it in production code.

Reply:
The annotation of `ResultPartitionState#EMPTY` is 
> Producer is not yet running.

I found that `RELEASED` is more proper than other enum value for `CANCELING`, `CANCELED`, `FAILED`, `RECONCILING`, so I return it.

What's your opinion of the mapping of the `ExecutionState` mentioned above ? Shall we return `ResultPartitionState#EMPTY` and modify the annotation to
> Producer is not yet running or in abnormal state.

Reply:
Javadoc for `EMPTY` should be rewritten – can be done in a follow up. I would say that `RELEASED` is a terminal state which is transitioned into as a result of a conscious decision to release the partition. Task failure should not result in a _"released"_ partition.

cc: @tillrohrmann

Reply:
I guess that in case of a failure we should return the partition state to the initial state (e.g. `EMPTY`).

Reply:
OK, in the latest PR, I return `EMPTY` for the failure state.